eth/stagedsync/stage_headers.go

```diff
@@ -110,6 +110,11 @@ func SpawnStageHeaders(
 		}
 		defer tx.Rollback()
 	}
+	if initialCycle {
```

Comment:
initialCycle == true, only for the first time?

Reply:
Yes, only for the first time, then will always be false afterwards

Reply:
Look through [the issue](https://github.com/ledgerwatch/erigon/issues/7339) please and [PR](https://github.com/ledgerwatch/erigon/pull/7532) this is crucial for lagging networks like BSC. You can find more at erigon's discord - https://discord.com/channels/687972960811745322/983710293156851763/1108134331882410045

Reply:
@alexqrid pls create an issue to track your issue. Your reply is irrelevant to this PR.